index.js

```diff
@@ -107,6 +106,7 @@ const ruleFunction = (expectation, options, context) => {
 
     let prettierSource;
 
+    const source = root.toString();
```

Comment:
This is where asynchrony comes into play. Previously, this code was executed before some asynchronous operations, and it turned out that first this plugin gets the code, after that other rules made changes to the code, and after that the prettier was launched on the old version of the code. Now this code works synchronously because it runs after all asynchronous operations.